lib/RestructureCFG/GenericRegionInfo.cpp

```diff
@@ -0,0 +1,200 @@
+//
+// Copyright rev.ng Labs Srl. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/DepthFirstIterator.h"
+#include "llvm/Analysis/CycleAnalysis.h"
+#include "llvm/IR/CFG.h"
+#include "llvm/IR/Function.h"
+
+#include "revng/Support/Debug.h"
+#include "revng/Support/GraphAlgorithms.h"
+
+#include "revng-c/RestructureCFG/GenericRegionInfo.h"
+
+using namespace llvm;
+
+// Debug logger
+Logger<> GenericRegionInfoLogger("generic-region-info");
+
+template<class GraphT, class GT>
+void GenericRegionInfo<GraphT, GT>::initializeRegions(GraphT F) {
+  // We instantiate the `GenericCycle` analysis and wrap the results in
+  // the region objects
+  CycleInfo CI;
+  CI.compute(*F);
+
+  using CycleT = CycleInfo::CycleT;
+  using Region = GenericRegion<NodeT>;
+  std::map<const CycleT *, Region *> CycleToRegionMap;
```

Comment:
Mmmm, maybe you meant `llvm::SmallDenseMap`? I assumed so, changed this here https://github.com/revng/revng-c/blob/f69bcce2af89d833714a56a52ae34f3581bae8f2/lib/RestructureCFG/GenericRegionInfo.cpp#L38

Also, since `llvm::SmallDenseMap` does not have the useful `at`, I added a simple standalone function to mimic the behavior here https://github.com/revng/revng-c/blob/f69bcce2af89d833714a56a52ae34f3581bae8f2/lib/RestructureCFG/GenericRegionInfo.cpp#L22

Reply:
Yeah, most of your maps were mapping pointers to pointers. `SmallDenseMap` FTW. Thanks